westcoast/src/main/java/frc/robot/Robot.java

```diff
@@ -96,42 +98,74 @@ public void teleopInit() {
                 .retrieveIntake
                 .whenPressed(retrieve(definition));
 
+
+        final Color teleopChosenColor;
         if (DriverStation.getAlliance() == DriverStation.Alliance.Blue) {
-            chosenColor = MagazineSubsystem.BlueTarget;
+            teleopChosenColor = MagazineSubsystem.BlueTarget;
         } else {
-            chosenColor = MagazineSubsystem.RedTarget;
+            teleopChosenColor = MagazineSubsystem.RedTarget;
         }
 
         definition
                 .input
                 .collect
-                .whileActiveOnce(collect(definition, () -> chosenColor));
+                .whileActiveOnce(collect(
+                        definition,
+                        () -> definition.subsystems.magazine.isMatchingColor(teleopChosenColor)
+                ));
+
+
+        definition
```

Comment:
This should probably be on a toggle and the smartdashboard should display what is selected (FAST or SLOW)

Reply:
I can agree with the Dashboard. I'll left it as a work for any student that want to put the information in the dashboard.
I would say we don't know if should be a toggle. I thought about this one for a while, and concluded it is better to put in different buttons so you only need to think I want fast or I want it slow. Using a toggle is more like check what is set then change if needed. So my choice ended in using 2 different buttons, but we can change during practice.

Reply:
I think using 2 different buttons makes the already completed control scheme even worse. However it is definitely something that should just be tested